test/e2e/tests/test_references.py

```diff
@@ -95,26 +93,30 @@ def referring_role(request, referred_policy_name):
         role.wait_until_deleted(role_name)
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="function")
 def referred_policy(request, referred_policy_name):
     policy_desc = "a referred-to policy"
 
     marker = request.node.get_closest_marker("resource_data")
     filename = "policy_simple"
     namespace = "default"
 
+    replacements = REPLACEMENT_VALUES.copy()
+
     if marker is not None:
         data = marker.args[0]
-        if 'withNamespace' in data:
+        if 'withNamespace' in data and data['withNamespace']:
+            logging.info("ChangingNamespaceInPolicy")
+            k8s.create_k8s_namespace(
+                TESTING_NAMESPACE
+            )
```

Comment:
Do you want to create the namespace at this level? why not at the beginning of the test function?

Reply:
I want to only create the namespace if needed..I will get `namespace already exists` error if it's created for the second test.
Also referred_policy runs before the test does

Reply:
fair enough, thank you!